server.js

```diff
@@ -1,32 +1,117 @@
 import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
+import "dotenv/config";
+import booksData from "./data/books.json";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost:27017/books";
+const connectToMongo = () => {
+  mongoose
+    .connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+    .then(() => console.log("Connected successfully"))
+    .catch((err) => {
+      console.error("Connection error:", err);
+      setTimeout(connectToMongo, 5000);
+    });
+};
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+connectToMongo();
 mongoose.Promise = Promise;
 
+//Creates a mongoose model named "Books"
+const Books = mongoose.model("Books", {
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_review_count: Number,
+});
+
 // Defines the port the app will run on. Defaults to 8080, but can be overridden
 // when starting the server. Example command to overwrite PORT env variable value:
 // PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
+const listEndpoints = require("express-list-endpoints");
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(listEndpoints(app));
+});
+
+// Route to get all books
+app.get("/books", (req, res) => {
+  try {
+    res.json(booksData);
+  } catch (error) {
+    res.status(500).json({ error: "Internal Server Error" });
+  }
+});
```

Comment:
This is nice looking code.. But I'm not sure the data is collected from the database, I think it might just be directly from the .json-file.

To get data into the database you need to seed it once, this is my code from last week: 
`const seedDataBase = async () => {
  data.forEach(song => {
    new ASong(song).save()
  })
}`
Where I stored my .json-file as the variable data. And my model was called ASong. So in your case you would use "booksData" for the json-file and Books for the model.

And once the data base was seeded, my .get-route looked like:
```try{
const songs = await ASong.find()
if (songs.length>1){
res.json({songs})
}
else {
res.status(404).json({error: "The database is empty, no songs found"})
}
catch (error) {
    res.status(500).json({error: "Something went wrong while loading the songs from the database"})
}```